lib/cfndsl/resources.rb

```diff
@@ -1,13 +1,12 @@
 require 'cfndsl/jsonable'
-require 'cfndsl/metadata'
 require 'cfndsl/properties'
 require 'cfndsl/update_policy'
 
 module CfnDsl
   # Handles Resource objects
   class ResourceDefinition < JSONable
-    dsl_attr_setter :Type, :DependsOn, :DeletionPolicy, :Condition
-    dsl_content_object :Property, :Metadata, :UpdatePolicy, :CreationPolicy
```

Comment:
I'm comfortable with removing this, I think @stevenjack your or statement makes sense, it's never really been relevant.